src/main/kotlin/org/rust/cargo/project/workspace/CargoProjectWorkspace.kt

```diff
@@ -1,23 +1,37 @@
 package org.rust.cargo.project.workspace
 
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.util.Disposer
+import com.intellij.util.messages.Topic
 import org.rust.cargo.project.CargoProjectDescription
 import org.rust.cargo.toolchain.RustToolchain
 
 /**
- * Uses `cargo metadata` command to update IDEA libraries and Cargo project model.
+ * Cargo based project's workspace abstraction insulating inter-op with the `cargo` & `Cargo.toml`
+ * itself. Uses `cargo metadata` sub-command to update IDEA module's & project's model.
+ *
+ * Quite low-level in its nature & follows a soft-fail policy, i.e. provides access
+ * for latest obtained instance of the [CargoProjectDescription], though doesn't assure that this
+ * one could be obtained (consider the case with invalid, or missing `Cargo.toml`)
  */
 interface CargoProjectWorkspace {
+
     /**
-     * Updates Rust libraries asynchronously. Consecutive updates are coalesced.
+     * Updates Rust libraries asynchronously. Consecutive requests are coalesced.
      */
-    fun scheduleUpdate(toolchain: RustToolchain)
+    fun requestUpdateUsing(toolchain: RustToolchain, immediately: Boolean = false)
 
     /**
-     * Immediately schedules an update. Shows balloon upon completion.
+     * Latest version of the Cargo's project-description obtained
      *
-     * Update is still asynchronous.
+     * NOTA BENE: In the current implementation it's SYNCHRONOUS
```

Comment:
Looks like the "SYNCHRONOUS" part is now misleading. We do not do any drastic blocking in this call any more (non EDT thread can block on the `lock` for a short period of time, ofc)